src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -1328,6 +1328,14 @@ type IncrementalBuilder(tcGlobals,frameworkTcImports, nonFrameworkAssemblyInputs
     let StampFileNameTask (cache: TimeStampCache) _ctok (_m:range, filename:string, _isLastCompiland) =
         assertNotDisposed()
         cache.GetFileTimeStamp filename
+
+    // Deduplicate module names
+    let seen = Dictionary<string,Set<string>>()
```

Comment:
@forki This duplicates the code added to fsc.fs?  Could you extract this to CompileOptions.fs or CompileOps.fs and share it please? https://github.com/Microsoft/visualfsharp/blob/f6e160a2e4a375754bcf62fc3641b1861093ca3b/src/fsharp/fsc.fs#L1768